### PR TITLE
Refactor key dispatcher tests - Pass 1

### PR DESCRIPTION
**Pull request by aicioara**
Part of #2628

I will be doing 2 passes because I will get more context (and better sense of best practices) by refactoring other files. However, please flag anything that seems unusual.

As part of this: 
- **Test semantics: hierarchy, naming, testing, but no coverage check**
- Applied best practices from the wiki page
- Factored out common code into `beforeEach()` clauses
- No nested `beforeEach()`es
- Cleaned the console
- Remade hierarchy as shown bellow

screen shot 2015-09-11 at 4 42 34 pm

### CODE REVIEW

File: test/dispatchers/keyDispatcherTests.ts [hunk 1]
```diff
@@ -2,50 +2,59 @@
...
+          assert.strictEqual(code, keyCodeToSend, "correct keycode was passed");
+          assert.isNotNull(event, "key event was passed to the callback");
+        };
+
+        assert.strictEqual(keyDispatcher.onKeyUp(callback), keyDispatcher, "setting the keyUp callback returns the dispatcher");
+
+        TestMethods.triggerFakeKeyboardEvent("keyup", body, keyCodeToSend);
+        assert.isTrue(keyUped, "callback when a key was release");
+
+        assert.strictEqual(keyDispatcher.offKeyUp(callback), keyDispatcher, "unsetting the keyUp callback returns the dispatcher");
```
Comment (bluong) [inline test/dispatchers/keyDispatcherTests.ts]: Can we separate the `offKeyUp` test from `onKeyUp`?

File: test/dispatchers/keyDispatcherTests.ts [hunk 2]
```diff
@@ -2,50 +2,59 @@
...
+
+        assert.strictEqual(keyDispatcher.onKeyDown(callback), keyDispatcher, "setting the keyDown callback returns the dispatcher");
 
-      ked.onKeyUp(callback);
+        TestMethods.triggerFakeKeyboardEvent("keydown", body, keyCodeToSend);
+        assert.isTrue(keyDowned, "callback when a key was pressed");
 
-      TestMethods.triggerFakeKeyboardEvent("keyup", body, keyCodeToSend);
-      assert.isTrue(keyUped, "callback when a key was release");
+        assert.strictEqual(keyDispatcher.offKeyDown(callback), keyDispatcher, "unsetting the keyDown callback returns the dispatcher");
```
Comment (bluong) [inline test/dispatchers/keyDispatcherTests.ts]: Can we separate the `offKeyDown` test from `onKeyDown`?

File: test/dispatchers/keyDispatcherTests.ts [hunk 3]
```diff
@@ -2,50 +2,59 @@
...
+
+      let body: d3.Selection<void>;
+
+      beforeEach(() => {
+        keyCodeToSend = 65;
+        keyDispatcher = Plottable.Dispatchers.Key.getDispatcher();
+        body = d3.select("body");
+      });
+
+      it("calls the onKeydown callback", () => {
```
Comment (bluong) [inline test/dispatchers/keyDispatcherTests.ts]: Maybe change the test name to something like
it("can set a callback to be called while a key is pressed down")?

Mainly change the name to be more semantic and more aligned to what a user is trying to do.

File: test/dispatchers/keyDispatcherTests.ts [hunk 4]
```diff
@@ -2,50 +2,59 @@
...
 
-      ked.offKeyUp(callback);
+        TestMethods.triggerFakeKeyboardEvent("keydown", body, keyCodeToSend);
+        assert.isFalse(keyDowned, "nothing happens when a key was pressed");
+      });
 
-      keyUped = false;
-      TestMethods.triggerFakeKeyboardEvent("keyup", body, keyCodeToSend);
-      assert.isFalse(keyUped, "nothing happens when a key was release");
+      it("calls the onKeyup callback", () => {
```
Comment (bluong) [inline test/dispatchers/keyDispatcherTests.ts]: Same as above.

File: test/dispatchers/keyDispatcherTests.ts (hunk 2 above)
Comment (aicioara) [inline test/dispatchers/keyDispatcherTests.ts]: done

File: test/dispatchers/keyDispatcherTests.ts [hunk 5]
```diff
@@ -2,50 +2,83 @@
...
+        assert.isTrue(keyDowned, "callback when a key was pressed");
+
+        keyDispatcher.offKeyDown(callback);
+      });
+
+      it("can remove the keydown listener", () => {
+        let keyDowned = false;
+        let callback = () => keyDowned = true;
+
+        assert.strictEqual(keyDispatcher.onKeyDown(callback), keyDispatcher, "setting the keyDown callback returns the dispatcher");
```
Comment (bluong) [inline test/dispatchers/keyDispatcherTests.ts]: Technically not necessary.  Not a big deal.

**Pull request merged**